### PR TITLE
minimacy: init at 0.6.2

### PR DESCRIPTION
**Pull request by jboynyc**
###### Description of changes

Adds a new package [minimacy](https://minimacy.net/), a compiler for an open-source minimalist computation system.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Hello, thanks for your contribution :) Could you add yourself as a maintainer for this package ? -> add yourself in the ``maintainers/maintainer-list.nix`` file and fill the ``meta.maintainer`` attribute.

File: pkgs/development/compilers/minimacy/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "minimacy";
+  version = "0.6.2";
+
+  src = fetchFromGitHub {
+    owner = "ambermind";
+    repo = "minimacy";
+    rev = "refs/tags/${version}";
```
Comment (JulienMalka) [inline pkgs/development/compilers/minimacy/default.nix:12]:
```suggestion
    rev =  version;
```

File: pkgs/development/compilers/minimacy/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+, alsaLib, libX11, libXext, libGL, libGLU
+}:
+
+stdenv.mkDerivation rec {
+  pname = "minimacy";
+  version = "0.6.2";
+
+  src = fetchFromGitHub {
+    owner = "ambermind";
+    repo = "minimacy";
```
Comment (JulienMalka) [inline pkgs/development/compilers/minimacy/default.nix:11]:
```suggestion
    repo = pname;
```

**Comment by jboynyc** [comment]
Thanks @JulienMalka for the guidance!

File: pkgs/development/compilers/minimacy/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  nativeBuildInputs = [ makeBinaryWrapper ];
+
+  buildInputs = [ libGL libGLU ] ++ lib.optionals stdenv.isLinux [ alsaLib libX11 libXext ];
+
+  enableParallelBuilding = true;
+
+  NIX_CFLAGS_COMPILE = "-Wno-unused-result";
+
+  buildPhase = ''
```
Comment (raphaelr) [inline pkgs/development/compilers/minimacy/default.nix:24]: add `runHook preBuild` at the beginning, and `runHook postBuild` at the end.

**Comment by flokli** [comment]
This mass-subscribed a bunch of people, who can't be unsubscribed. Please open a new PR, closing this one to prevent more noise.

**Pull request closed without merging**